### PR TITLE
Define a module for home

### PR DESCRIPTION
**Pull request by gmarziou**
Home page should have its own module to keep app.module simple and high level and because it gives some structure to users for modifying it.

Remove layout block dependency on home route as it deals only with layout not content.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { ParseLinks } from 'ng-jhipster';
+
+import { ParattSharedModule } from '../shared';
+
+import { HOME_ROUTES, HomeComponent } from './';
+
+
+@NgModule({
+    imports: [
+        ParattSharedModule,
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts]: Missing template here

Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts]: AppName need to be in template

**Comment by gmarziou** [comment]
Travis errors unrelated to code change

**Comment by jdubois** [comment]
@gmarziou I'm restarting Travis -> you don't have permission to do it? You should, as part of the core dev team, but I have no idea where this is configured!

File: generators/client/files-angular.js [hunk 2]
```diff
@@ -126,6 +126,7 @@ const files = {
                 'entities/_entity.module.ts',
                 // home module
                 'home/_index.ts',
+                { file: 'home/_home.module.ts', method: 'copyJs' },
```
Comment (deepu105) [inline generators/client/files-angular.js]: no need for copyJs as its required only if we have any i18n regex to strip. may be I should rename the method names to be more meaningful

File: generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { RouterModule } from '@angular/router';
+import { ParseLinks } from 'ng-jhipster';
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts]: I dont think we are using parseLinks here?

File: generators/client/templates/angular/src/main/webapp/app/home/_home.route.ts [hunk 4]
```diff
@@ -1,14 +1,16 @@
-import { Route } from '@angular/router';
+import { Routes } from '@angular/router';
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.route.ts]: any reason to change to Route array?

Comment (gmarziou) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.route.ts]: That's what I've seen in some angular examples and it looked nicer to read but I can change if you prefer the array.

File: generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts (hunk 3 above)
Comment (gmarziou) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.module.ts]: Correct, I'll remove it

File: generators/client/files-angular.js (hunk 2 above)
Comment (gmarziou) [inline generators/client/files-angular.js]: OK, I'll have a look.

File: generators/client/templates/angular/src/main/webapp/app/home/_home.route.ts (hunk 4 above)
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/home/_home.route.ts]: we are using Route everywhere except when we have multiple routes in same file

**Comment by gmarziou** [comment]
Closing to replace by corrected one.

**Pull request closed without merging**